### DISCUSSION TITLE
Deployment error  with setup complete

### DISCUSSION THREAD

**Discussion by gyokiss** [quay/quay#2187] [Q&A]
Hi,

I have started to gather some experiences qith quay. My first step as usual, follow the doc and deploy a system. Now do I have a little issue: 

`
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
Running init script '/quay-registry/conf/init/supervisord_conf_create.sh'
Running init script '/quay-registry/conf/init/zz_boot.sh'
Traceback (most recent call last):
  File "/quay-registry/boot.py", line 111, in <module>
    main()
  File "/quay-registry/boot.py", line 98, in main
    raise Exception(

Exception: Your configuration bundle is either not mounted or setup has not been complete`

After some googleing, I have found only advice: use the latest... 

The easy solution  - add a setup_complete true to config file ofc not helped. 

Where can I move forward, where are some logs? 

Thanks
Gyokiss

**Comment by ibazulic** [comment]
Hey there,

How are you deploying your Quay? You should never add the `SETUP_COMPLETE` flag manually.

**Reply by gyokiss** [reply]
I gave a try to follow the docu: https://docs.projectquay.io/deploy_quay.html#_overview

postgres done, redis done, config start fone, validaton looks fine. 
Than start with: 
`sudo podman run --rm -p 80:8080 -p 443:8443 --name=quay -v $QUAY/config:/conf/stack:Z -v $QUAY/storage:/datastorage:Z quay.io/projectquay/quay`

I have removed the  "d" flag, to see what happens...

**Reply by gyokiss** [reply]
It looks like in my config file missing the DATABASE_SECRET_KEY and the SECRET_KEY entries...

**Reply by ibazulic** [reply]
The configuration is not complete. Did you start the config tool and proceeded to configure Quay via its web interface?

**Reply by gyokiss** [reply]
i had started
 `[quay@quay ~]$ sudo podman run --rm -it --name quay_config -p 80:8080 -p 443:8443 quay.io/projectquay/quay:v3.8.0 config secret`
than configure the postgres, redis and server name.
Validation, and downloaded the config file. 
Here I gave a try to start ... and this is where I'm now.

**Comment by gyokiss** [comment]
Hi 
now its better i had made again the config run, but now from a fedora box ( not my ususal windows workstation) - it looks better, but not yet fully finised. 
The database secret is in config file.

The interesting behavior: my session constantly redirected to config site. hm. not really got why. I would expect the possibility to add a new user...
Is there any method to add user in cli?

**Comment by ibazulic** [comment]
The `config.yaml` file must contain both the `SECRET_KEY` and `DATABASE_SECRET_KEY` in addition to the storage setup, auth setup and a bunch of other stuff. The `SETUP_COMPLETE: true` should be added on validation. I'm not sure why are you using version 3.8.0, perhaps something went wrong with that version config tool, not sure. I haven't really tried. I would recommend you use the latest stable:

~~~
quay.io/projectquay/quay:v3.9.0
~~~

which was released 15 days ago. As far as creation of users from CLI is concerned, there's an init API that you can call after Quay has been bootstrapped for the first time. To make it work, add the following to the `config.yaml` file:

~~~
FEATURE_USER_INITIALIZE: true
~~~

API is called in the following manner:

~~~
# curl -X POST -H "Content-type: application/json" -d '{"username":"username", "password":"your_password", "email":"email@email.com"}' https://QUAY_HOSTNAME/api/v1/user/initialize
{"email":"email@email.com","encrypted_password":"Z1T2X...","username":"username"}
~~~

Alternatively, you do it via the UI. There should be a "Create account" link on the landing page. I'm not entirely sure why are you being redirected to the config tool, that seems to be something connected to your browser.

HTH!
